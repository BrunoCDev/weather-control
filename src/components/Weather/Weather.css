.weather-container {
  width: 100%;
  height: 100%; }

.weather-header {
  height: 10%;
  width: 100%; }

.weekday-text {
  color: #adb9c1;
  font-size: 100%;
  text-align: center;
  vertical-align: middle;
  line-height: 0%; }

.weather-body {
  height: 84%;
  width: 100%;
  text-align: center;
  vertical-align: middle; }

.icon {
  width: 60%;
  height: 70px; }

.description-text {
  color: #ffffff;
  font-size: 90%; }

.temp-cel {
  color: #ffffff;
  font-weight: bold;
  font-size: 80%; }

.temp-fel {
  color: #ffffff;
  text-align: center;
  font-size: 70%;
  line-height: 0%; }
