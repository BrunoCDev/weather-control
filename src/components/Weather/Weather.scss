.weather-container {
  background-color: #1f212b;
}

.weather-header {
  background-color: #1f212b;
  height: 10%;
}

.weekday-text {
  color: #adb9c1;
  line-height: 50%;
  font-size: 50%;
  text-align: center;
  vertical-align: middle;
}

.weather-body {
  height: 90%;
  width: 100%;
  background-color: #2c2e3c;
  text-align: center;
  vertical-align: middle;
}

.icon {
  margin-top: 10%;
}

.description-text {
  color: #ffffff;
  font-size: 50%;
}

.temp-cel {
  color: #ffffff;
  font-weight: bold;
  font-size: 45%;
}

.temp-fel {
  color: #ffffff;
  text-align: center;
  font-size: 25%;
  line-height: 10%;
}
