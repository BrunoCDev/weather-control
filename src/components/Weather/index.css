.weather-container {
  width: 20vw;
  height: 32vh; }
  @media (min-width: 0px) {
    .weather-container {
      height: 160px;
      width: 250px; } }
  @media (min-width: 550px) {
    .weather-container {
      height: 230px;
      width: 25%; } }
  @media (min-width: 800px) {
    .weather-container {
      height: 320px;
      width: 180px; } }

.weather-header {
  height: 3.5vh;
  width: 20vw; }
  @media (min-width: 0px) {
    .weather-header {
      height: 30px;
      width: 62px; } }
  @media (min-width: 550px) {
    .weather-header {
      height: 50px;
      width: 120px; } }
  @media (min-width: 800px) {
    .weather-header {
      height: 60px;
      width: 180px; } }

.weekday-text {
  color: #adb9c1;
  font-weight: bold;
  letter-spacing: 0.5px;
  text-align: center;
  vertical-align: middle;
  margin: 0px; }
  @media (min-width: 0px) {
    .weekday-text {
      height: 30px;
      width: 62px;
      font-size: 10px;
      line-height: 30px; } }
  @media (min-width: 550px) {
    .weekday-text {
      height: 50px;
      width: 120px;
      font-size: 15px;
      line-height: 50px; } }
  @media (min-width: 800px) {
    .weekday-text {
      height: 60px;
      width: 180px;
      font-size: 20px;
      line-height: 60px; } }

.weather-body {
  height: 27.5vh;
  width: 100%;
  text-align: center;
  vertical-align: middle; }
  @media (min-width: 0px) {
    .weather-body {
      height: 130px; } }
  @media (min-width: 550px) {
    .weather-body {
      height: 230px; } }
  @media (min-width: 800px) {
    .weather-body {
      height: 260px; } }

.icon {
  width: 15vw;
  height: auto; }
  @media (min-width: 0px) {
    .icon {
      width: 40px; } }
  @media (min-width: 550px) {
    .icon {
      width: 80px; } }
  @media (min-width: 800px) {
    .icon {
      width: 80px; } }

.description-text {
  color: #ffffff;
  width: 100%;
  height: 2vh;
  font-size: 3vw;
  letter-spacing: 0.7px; }
  @media (min-width: 0px) {
    .description-text {
      height: 12px;
      font-size: 9px; } }
  @media (min-width: 550px) {
    .description-text {
      height: 20px;
      font-size: 15px; } }
  @media (min-width: 800px) {
    .description-text {
      height: 20px;
      font-size: 22px; } }

.temp-cel {
  color: #ffffff;
  font-size: 5vw;
  letter-spacing: 1.5px; }
  @media (min-width: 0px) {
    .temp-cel {
      font-size: 15px; } }
  @media (min-width: 550px) {
    .temp-cel {
      font-size: 20px; } }
  @media (min-width: 800px) {
    .temp-cel {
      font-size: 30px; } }

.temp-fel {
  color: #ffffff;
  text-align: center;
  font-size: 3vw;
  line-height: 0px;
  letter-spacing: 0.5px; }
  @media (min-width: 0px) {
    .temp-fel {
      font-size: 9px; } }
  @media (min-width: 550px) {
    .temp-fel {
      font-size: 15px; } }
  @media (min-width: 800px) {
    .temp-fel {
      font-size: 20px; } }
