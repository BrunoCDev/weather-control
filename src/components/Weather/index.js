import React, { Component } from "react";
import "./index.css";

class Weather extends Component {
  constructor(props) {
    super(props);
    this.state = {
      week: [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday"
      ],
      loading: true
    };
    this.changeColor = this.changeColor.bind(this);
    this.parseDescription = this.parseDescription.bind(this);
    this.parseTemp = this.parseTemp.bind(this);
  }

  componentDidMount() {
    this.setState({ loading: false });
  }

  // Changes the <div> color to create pattern.
  changeColor(i, str) {
    if (i % 2 === 0 && str === "header") {
      return "#282936";
    }
    if (i % 2 !== 0 && str === "header") {
      return "#1F212B";
    }
    if (i % 2 === 0 && str === "body") {
      return "#2C2E3C";
    } else {
      return "#222530";
    }
  }

  // Uppercases the first letter of the description.
  parseDescription(description) {
    description = description.split(" ");
    const result = [];
    description.map(el => {
      result.push(el[0].toUpperCase() + el.slice(1, el.length));
    });
    return result.join(" ");
  }

  // Converts temperature received by API (Kelvin) to Celsius or Fahrenheit.
  parseTemp(temp, str) {
    if (str === "C") {
      return `${Math.round(temp - 273.15)}°C`;
    }
    if (str === "F") {
      return `${Math.round((temp - 273.15) * 1.8 + 32)}°F`;
    }
  }

  render() {
    // Object destructuring for shorthand.
    const data = this.props.data;
    const { icon, description } = data.info.weather[0];
    const { temp } = data.info.main;
    const { week, loading } = this.state;
    const { index } = this.props;

    // Gets the day number.
    const date = data.info.dt_txt.split(" ");
    const day = date[0].split("-");

    // Conditional Render so the component changes the displayed information after
    // the information changes.
    return (
      !loading && (
        <div
          className="weather-container"
          style={{ backgroundColor: this.changeColor(index, "header") }}
        >
          <div
            className="weather-header"
            style={{ backgroundColor: this.changeColor(index, "header") }}
          >
            <p className="weekday-text">
              {index === 0
                ? "Today " + day[2]
                : week[data.weekDay] + ` ${day[2]}`}
            </p>
          </div>
          <div
            className="weather-body"
            style={{ backgroundColor: this.changeColor(index, "body") }}
          >
            <div>
              <img
                className="icon"
                src={`http://openweathermap.org/img/w/${icon}.png`}
                alt="Weather"
              />
            </div>
            <div>
              <p className="description-text">
                {this.parseDescription(description)}
              </p>
            </div>
            <div>
              <p className="temp-cel">{this.parseTemp(temp, "C")}</p>
            </div>
            <div className="temp-fel-container">
              <p className="temp-fel">{this.parseTemp(temp, "F")}</p>
            </div>
          </div>
        </div>
      )
    );
  }
}

export default Weather;
