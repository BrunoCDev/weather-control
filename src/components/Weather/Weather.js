import React, { Component } from "react";
import "./Weather.css";

class Weather extends Component {
  constructor(props) {
    super(props);
    this.state = {
      week: [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday"
      ]
    };
  }
  render() {
    console.log(this.props.data);
    const data = this.props.data;
    const { icon, description } = data.info.weather[0];
    const { temp } = data.info.main;
    const { week } = this.state;
    return (
      <div className="weather-container">
        <div className="weather-header">
          <p className="weekday-text">{week[data.weekDay]}</p>
        </div>
        <div className="weather-body">
          <div>
            <img
              className="icon"
              src={`http://openweathermap.org/img/w/${icon}.png`}
              alt="Weather"
            />
          </div>
          <div>
            <p className="description-text">{description}</p>
          </div>
          <div>
            <p className="temp-cel">{Math.round(temp - 273.15)}°C</p>
          </div>
          <div>
            <p className="temp-fel">
              {Math.round((temp - 273.15) * 1.8 + 32)}°F
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default Weather;
