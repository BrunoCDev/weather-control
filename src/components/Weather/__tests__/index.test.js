import React from "react";
import ReactDOM from "react-dom";
import Enzyme from "enzyme";
import { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

import Weather from "./../index";

it("Renders components without crashing", () => {
  const renderedComponent = mount(<div>Test</div>);
  expect(renderedComponent).toMatchSnapshot();
});

describe("<Weather/>", () => {
  it("Should render nested components", () => {
    const children = <div />;
    const renderedComponent = mount(<div>{children}</div>);
    expect(renderedComponent.contains(children)).toEqual(true);
  });

  it("Should render an <img> tag with className", () => {
    const renderedComponent = mount(<img className="icon" alt="Weather" />);
    expect(renderedComponent.find("img").hasClass("icon")).toEqual(true);
    expect(renderedComponent).toMatchSnapshot();
  });

  it("Should adopt the className", () => {
    const renderedComponent = mount(<div className="weather-container" />);
    expect(renderedComponent.find("div").hasClass("weather-container")).toEqual(
      true
    );
  });
});
