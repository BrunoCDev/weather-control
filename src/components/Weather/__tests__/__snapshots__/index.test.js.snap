// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather/> Should render an <img> tag with className 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <img
    alt="Weather"
    className="icon"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <img
      class="icon"
      alt="Weather"
/>,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "Weather",
      "className": "icon",
    },
    "ref": null,
    "rendered": Array [
      undefined,
    ],
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <img
        class="icon"
        alt="Weather"
/>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Weather",
        "className": "icon",
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Renders components without crashing 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <div>
    Test
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <div>
      Test
</div>,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test",
    },
    "ref": null,
    "rendered": Array [
      "Test",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <div>
        Test
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
      },
      "ref": null,
      "rendered": Array [
        "Test",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
